### ISSUE TITLE
Owncloud ubuntu package: php5 dependency

### ISSUE THREAD

**Issue by Pegasuz** [owncloud/core#20963]
I am using the ubuntu 14.04 package on my server and would like to upgrade to php7 (currently using php5.6 mod_apache).
The owncloud-server package (version 8.2.1) however depends on php5.
So it is currently not possible to upgrade to php7.

**Comment by Xenopathic** [comment]
cc @jnweiger

**Comment by Akhenaton** [comment]
:~$ apt-cache depends owncloud-server
owncloud-server
  Dépend: php5
 |Dépend: php5-mysqlnd
  Dépend: php5-mysql
  Dépend: php5-sqlite
  Dépend: php5-pgsql
  Dépend: php5-common
  Dépend: php5-gd
  Dépend: php-xml-parser
  Dépend: php5-intl
  Dépend: php5-mcrypt
  Dépend: php5-curl
  Dépend: php5-json
  Dépend: php5-ldap
  Dépend: php5-cli

**Comment by enoch85** [comment]
I have the same problem. One way to solve this could be: http://askubuntu.com/a/237354 But I have not tested that solution yet. If anyone does, please let me know how it went.

**Comment by jnweiger** [comment]
Redis is even difficult with php5: the ubuntu-14.04 php5-redis module is too old, the say:
http://www.kiloroot.com/a-guide-to-owncloud-8-ubuntu-14-04-and-caching-apcu-redis-cache-and-file-locking-oh-my/

**Comment by enoch85** [comment]
@jnweiger That's why you have to build it: https://www.en0ch.se/how-to-configure-redis-cache-in-ubuntu-14-04-with-owncloud/

**Comment by jnweiger** [comment]
I've started somewhat. Feel free to improve: https://github.com/owncloud/vm/commit/252ddbdf6a9768803ff54c54b4e82f6b7ac1aed6

**Comment by enoch85** [comment]
@jnweiger If you want to install an old redis server version, that looks fine. :+1:

**Comment by jnweiger** [comment]
It's a production system. Compilers should run elsewhere.

**Comment by enoch85** [comment]
@jnweiger What about the original issue ---^ ?

**Comment by Xenopathic** [comment]
Please don't add a dependency on php-redis on owncloud-server, at most it should be an optional dependency, definitely not required or recommended

**Comment by karlitschek** [comment]
Yes. This is optional and only important for big installations.

**Comment by Akhenaton** [comment]
Well, yes, i should have said "optional" or "recommended" only. I have Redis server v=3.0.5 here and it works like a charm with ownCloud.

**Comment by enoch85** [comment]
FYI:
```
root@owncloud:/etc/redis# apt-mark hold owncloud owncloud-server owncloud-config-apache
owncloud set on hold.
owncloud-server set on hold.
owncloud-config-apache set on hold.
root@owncloud:/etc/redis# apt-get install php7.0
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 owncloud-server : Depends: php5 (>= 5.4.0) but it is not going to be installed
                   Depends: php5-mysqlnd but it is not going to be installed or
                            php5-mysql but it is not going to be installed
                   Depends: php5-sqlite but it is not going to be installed
                   Depends: php5-pgsql but it is not going to be installed
                   Depends: php5-common but it is not going to be installed
                   Depends: php5-gd but it is not going to be installed
                   Depends: php5-intl but it is not going to be installed
                   Depends: php5-mcrypt but it is not going to be installed
                   Depends: php5-curl but it is not going to be installed
                   Depends: php5-json but it is not going to be installed
                   Depends: php5-ldap but it is not going to be installed
                   Depends: php5-cli but it is not going to be installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.
root@owncloud:/etc/redis# 
```

**Comment by jnweiger** [comment]
The correct solution is to have an owncloud package without any dependencies. this may be helpful with many other dependency issues too.
I'll add an experimental package owncloud-base to https://download.owncloud.org/download/repositories/8.2:/testing/owncloud-base

**Comment by enoch85** [comment]
If you decide to go that path it has to be clear in the docs which dependencies that are needed to make ownCloud run properly. cc @carlaschroder

**Comment by enoch85** [comment]
1. Removed ownCloud
2. Deleted ownCloud dir inlcuding the data folder
3. Replaced owncloud.list with the new one.
4. Installed php7.0*

**Actual result**

```
root@owncloud:/# apt-get install owncloud-base 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 owncloud-base : Depends: php5-cgi but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
root@owncloud:/# 
```

**Comment by enoch85** [comment]
The VM is released, thought you would like to know: https://www.en0ch.se/owncloud-8-2-1-with-php-7-and-mysql-5-6/

**Comment by jospoortvliet** [comment]
btw I just installed php 7 + all the needed php 7 modules on openSUSE 13.2. It overwrote a few files due to conflicts but otherwise, everything was smooth and worked after I removed apcu caching from the config.php. I still have php5 installed, will remove it once ownCloud-server won't also be removed ;-)

**Comment by MrGraversen** [comment]
Well, so much for "PHP 7 Is Here And ownCloud Is Ready"... any news on the issue?

**Comment by enoch85** [comment]
@MrGraversen ownCloud works with PHP 7, just download the VM and give it a try. ;) You could get it working with ownCloud with some hacking in Ubuntu.

**Comment by enoch85** [comment]
@jnweiger Saw that the new 8.2.2 installed php5-cgi separately. Does this mean that the issue is addressed?

**Comment by Akhenaton** [comment]
No... 
Do an "apt-cache depends owncloud-server" => Still depends  PHP5- exclusively.

**Comment by MrGraversen** [comment]
Any news on the issue...? This is handled shockingly bad.

**Comment by jospoortvliet** [comment]
@MrGraversen right now, most people who work on this are on holiday. I'm sure there will be attention for this in the new year ;-)

**Comment by webmarka** [comment]
Got it working pretty easily using @enoch85 method on a Linux Mint 17.3 desktop. It's a basic install, I will keep testing with different configs and apps enabled too see if my favourites apps setup still work. If everything is good, next step will be to test it on a Ubuntu 14.04 server :) Yay!

**Comment by Akhenaton** [comment]
Look, we all know that it's working well... That's not the matter. The matter is a Linux packaging issue.

**Comment by jospoortvliet** [comment]
I believe the packages were updated to not handle dependencies at all... Not sure if that has already been applied fully, but it's a rather dirty solution to this problem.

What's going to happen with the packages is a tad unclear at the moment, we're not happy with the state either.

**Comment by Akhenaton** [comment]
> I believe the packages were updated to not handle dependencies at all...

No, it hasn't been updated... Perhaps for the next 8.2.3 ?... in March 2016.... pff

**Comment by enoch85** [comment]
@Akhenaton Just to be clear - ownCloud works with PHP 7. I've tested it, @jospoortvliet tested it - and I'm sure several others did to. I agree that the dependency issue is bad, but it's going to be solved.

**Comment by Akhenaton** [comment]
@enoch85 No, it doesn't !.... If you can't sync with a client, if you can't use caldav, if you can't use cardav, in other words if "dav" isn't working i don't call it "operational".

https://github.com/owncloud/calendar/issues/1002
https://github.com/owncloud/core/issues/21476
https://github.com/owncloud/core/issues/21460
https://forum.owncloud.org/viewtopic.php?f=36&t=32439

**Comment by enoch85** [comment]
@Akhenaton Well, I use all the clients for desktop, iOS Android and PC. I have no issues...

**Comment by Akhenaton** [comment]
@enoch85 O.K. good to know. But what about Caldav & Carddav ? I mean these last two are quite essentials for a lot of end-users like me.

**Comment by enoch85** [comment]
@Akhenaton As I wrote earlier:

> iOS, Android

And I use calendar (CalDAV), and contacts (CarDDAV) on those devices - and it works.

**Comment by SMillerDev** [comment]
Isn't this completely besides the point? The package depends on php5-json while it can depend on php-json. No version needed. Doesn't really matter if it works with it or not. It should be an option.


 Caldav and cardav problems are reported in different issues. And there's already fixes for those. 




On Sat, Jan 9, 2016 at 6:21 AM -0800, "Daniel Hansson" <notifications@github.com> wrote:












@Akhenaton As I wrote earlier:




iOS, Android




And I use calendar (CalDAV), and contacts (CarDDAV) on those devices - and it works.

**Comment by enoch85** [comment]
@Akhenaton I the new VM I made ownCloud is installed with PHP 7 and MySQL 5.6. Just give it a try and see what you think: https://en0ch.se/index.php/s/zfVYCMabgWdw90k

**Comment by Akhenaton** [comment]
@enoch85 Hi, thank you, i downloaded the VM and i will give it a try this week for sure. :-)
Not much time today... Again, thank you and... Happy new year !

**Comment by RealRancor** [comment]
Until https://github.com/owncloud/core/issues/21476 is solved no dependency update should be done as SabreDAV 3.0 currently used by oC is not fully supporting PHP7:  https://github.com/owncloud/core/issues/21476#issuecomment-170441859

**Comment by Akhenaton** [comment]
@RealRancor That's exactly what I thought, too ! Thanks for this clear statement. :-)

**Comment by carlaschroder** [comment]
PHP 7 is still experimental. I'm not documenting it until it it is available in most distros and the kinks are worked out. It would be awesome if the people who are using it would start a Wiki page on their experiences https://github.com/owncloud/documentation/wiki

**Comment by carlaschroder** [comment]
And there is an admin manual page that lists required, recommended, and optional PHP modules, plus other prerequisites https://doc.owncloud.org/server/8.2/admin_manual/installation/source_installation.html#prerequisites

**Comment by mvl22** [comment]
> PHP 7 is still experimental.

No, it's not. It's officially released, having been through an large number of betas and Release Candidates.

https://wiki.php.net/todo/php70#timetable

**Comment by carlaschroder** [comment]
PHP 7 support in ownCloud is still experimental.

**Comment by mvl22** [comment]
> PHP 7 support in ownCloud is still experimental.

Thanks for the clarification.

However, that should probably be stated here if so:
https://owncloud.org/blog/php-7-is-here-and-owncloud-is-ready/
as there is no such indication.

Looking forward to the packaging issue noted above being solved and put into the main official release. Thanks for the hard work on that, to those that are working on it.

**Comment by Akhenaton** [comment]
@mvl22 I agree, this announce "ownCloud ready for php7" dated *December 3, 2015*, was misleading many users. And i must say the @enoch85 posts here had made me think everything was O.K with php7....  :confused:

**Comment by RealRancor** [comment]
cc @jospoortvliet

**Comment by enoch85** [comment]
@Akhenaton Have you tried the VM I linked? Please do, and give it a thorough test, and then write here if you think PHP 7 is working with ownCloud or not. Here is the link again: https://en0ch.se/index.php/s/zfVYCMabgWdw90k

Just for the record: http://sabre.io/blog/2016/sabre-vobject-3.4.9-and-3.5.0-release/

And yes, some apps are not compatible with PHP 7 yet, but ownCloud is, IMHO.

**Comment by RealRancor** [comment]
@enoch85 SabreDAV 3.0 used by ownCloud is not compatible to PHP7 according to its author: https://github.com/owncloud/core/issues/21476#issuecomment-170441859 I think that guy should know this best. :-)

**Comment by enoch85** [comment]
@RealRancor Yes, but SabreDAV will be updated in oC 9 right?

**Comment by enoch85** [comment]
@Akhenaton Ok, so I did half of the work for you. ownCloud with PHP 7 is now on a public domain: https://owncloudphp7.techandme.se

USER: test_user
PASSWORD: testingphp7

Please connect with all the clients possible, and give it a good test! 

I will remove it if I notice any abuse.

**Comment by Xenopathic** [comment]
@enoch85 @Akhenaton Just performed some basic tests on that test server with the sync client and CalDAV via Thunderbird, all seemed to work fine. I'm feeling more confident upgrading now...

**Comment by enoch85** [comment]
@Xenopathic Thanks for that. Please continue to test if you want.

**Comment by Xenopathic** [comment]
@enoch85 Thanks for providing a test instance, it was really convenient :smile:

**Comment by enoch85** [comment]
@Xenopathic Glad I could help. 

Just tested the sync client with [this stress test](https://www.techandme.se/stress-test/). It took 10 minutes on PHP 5.5 on a quad core server. With PHP 7 it took 4 minutes on this dual core server. WOW, that's fast.

![php7](https://cloud.githubusercontent.com/assets/4511254/12248457/d1c55a3c-b8b9-11e5-9aa7-8d1db1cd0182.png)

**Comment by enoch85** [comment]
Ok, found a thing that breaks. It works to sync from phone to OCcalendar, but not from OCcalendar to phone.

```
{"reqId":"03zOFY1cKuoynE9z\/gLz","remoteAddr":"192.168.1.201","app":"PHP","message":"Array to string conversion at \/var\/www\/html\/owncloud\/3rdparty\/sabre\/dav\/lib\/CalDAV\/CalendarQueryValidator.php#62","level":3,"time":"2016-01-12T01:19:05+00:00"}
```

Other than that, all the tests I made worked. That is WebDAV, CalDAV, CarDDAV.

**Comment by Akhenaton** [comment]
@enoch85 Thank you to have put on line your test site. Helpfull !  :smiley:
I've done very quickly (lack of time) one or two tests -cardDAV, calDAV, and it's good for me.
I regret though the confusion made for end-user that had to wait more than a full month to get a clear statement about the issue...  :disappointed:
Keep the good work going strong ! Cheers

**Comment by enoch85** [comment]
@Akhenaton Sorry, just deleted the user, use `testuser` instead.

**Comment by enoch85** [comment]
@Akhenaton Did you have any time to test it yet? :)

**Comment by DeepDiver1975** [comment]
> @enoch85 SabreDAV 3.0 used by ownCloud is not compatible to PHP7 according to its author: #21476 (comment) I think that guy should know this best. :-)

sabre dav 3.0.7 has been release which fixes php7 compat

**Comment by Akhenaton** [comment]
@enoch85 Well, i got your VMware VM and will try today. But i use Oracle **Virtualbox** (last version 5.0.12) and it seems VirtualBox and VMware both seem to use slightly different OVA/OVF implementations that aren’t entirely compatible... I have also some problems to put my wifi in bridge mode... Well i will try to load directly the vmdk (without using the vmx) and install my virtual machine’s guest operating system from scratch.  :pray:

**Comment by enoch85** [comment]
@Akhenaton Here you go: https://www.techandme.se/complete-install-instructions-owncloud/

**Comment by MrGraversen** [comment]
Still waiting to install this on my Ubuntu server. Any news?

**Comment by RealRancor** [comment]
@MrGraversen No need to hurry as there are known issues with oC8.x and PHP7. You should wait for oC9 anyway :-)

**Comment by enoch85** [comment]
OC 9 is soon to be released and I don't know if this issue will be resolved by then. But installing current version under PHP 7 gives errors:

```
# Download and install ownCloud
wget -nv https://download.owncloud.org/download/repositories/stable/Ubuntu_14.04/Release.key -O Release.key
apt-key add - < Release.key && rm Release.key
sh -c "echo 'deb http://download.owncloud.org/download/repositories/stable/Ubuntu_14.04/ /' >> /etc/apt/sources.list.d/owncloud.list"
apt-get update && apt-get install owncloud -y
```
Gives this:
```
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 owncloud : Depends: owncloud-server (= 8.2.2-1.1) but it is not going to be installed
            Recommends: mysql-server but it is not going to be installed or
                        mariadb-server but it is not going to be installed
            Recommends: owncloud-config-apache (= 8.2.2-1.1) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

From [this](https://github.com/enoch85/ownCloud-VM/blob/production_rework/production/owncloud_install_production.sh#L137-L141) script

**Comment by enoch85** [comment]
@jnweiger I really thought this would be resolved by now. Trying to install ownCloud on a system with PHP 7 **still** fails. :(

```
The following packages have unmet dependencies:
 owncloud : Depends: owncloud-deps-php5 (>= 9.0.0) but it is not going to be installed or
                     owncloud-deps
E: Unable to correct problems, you have held broken packages.
```

Please tell me if you need any help.

**Comment by jnweiger** [comment]
With 9.0.0.0 this is fixable in two ways:

a) Install owncloud-files instead of owncloud -- and do the rest manually.
b) An owncloud-deps-php7 can be written to pull the needed php7 dependencies.

First someone would need to do a) and document the steps, based on this, we can work on b).

**Comment by enoch85** [comment]
@jnweiger Please talk to me on IRC, I'm idle in #owncloud-dev

**Comment by enoch85** [comment]
@jnweiger Btw, that could be documented on the download site. I can't edit that one, sorry.

But anyway, I'll try `owncloud-files` and do the rest manually. And by manually you mean the Apache config right?

**Comment by jkroepke** [comment]
Currently, owncloud-deps-php7 has wrong dependencies

```
# apt-get install owncloud-deps-php7
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 owncloud-deps-php7 : Depends: php7 but it is not installable
E: Unable to correct problems, you have held broken packages.
```

On Ubuntu 16.04 LTS, there is only a [php7.0](http://packages.ubuntu.com/xenial/any/php7.0) package, like in [debian](https://packages.debian.org/sid/php7.0) or in the [ppa from Ondřej Surý ](https://launchpad.net/~ondrej/+archive/ubuntu/php?field.series_filter=trusty)

owncloud-deps-php7 should be depend to php7.0 instead php7.

Hopefully, this gets fixed before 9.0.1 release.

**Comment by enoch85** [comment]
@jknockaert Thanks! Though you won't need it to install dependencies if you [pre install everything](https://github.com/enoch85/ownCloud-VM/blob/master/production/owncloud_install_production.sh#L114-L135).

**Comment by jkroepke** [comment]
Do you know when it will be fixed?

**Comment by enoch85** [comment]
Depends on how busy @jnweiger is. I hope that this will get in 9.0.1.

**Comment by jkroepke** [comment]
Thanks

**Comment by jnweiger** [comment]
This is the control file for the deps. Suggestions welcome:
<pre>
Package: owncloud-deps-php5
Architecture: all
Provides: owncloud-deps
# https://github.com/owncloud/core/issues/21342
Depends: ${misc:Depends}, php5 (>= 5.4.0),
        php5-common, php5-gd, php-xml-parser, php5-intl, php5-mcrypt, php5-curl, php5-json,
        php5-mysqlnd | php5-mysql, php5-sqlite, php5-pgsql,
        php5-cgi | php5-fpm | libapache2-mod-php5,
        apache2 (>= 2.4.0) | httpd
# Don't recommend php5-apcu for Ubuntu-14.04 https://github.com/owncloud/core/issues/14188
# Don't use apcu before 4.0.6 https://github.com/owncloud/core/issues/14175
# php5-imagick is needed by apps/files-versions
# php5-ldap is needed by apps/user-ldap
Suggests: php5-apcu (>= 4.0.6) | php-apc, php5-imagick, php5-ldap
Recommends: mysql-server | mariadb-server | postgresql, curl, exim4 | mail-transport-agent
Description: ownCloud - dependencies for php5
 .

Package: owncloud-deps-php7
Architecture: all
Provides: owncloud-deps
Depends: php7 | php7.0
Description: ownCloud - dependencies for php7
 .

</pre>
I've changed ``Depends: php7`` into ``Depends:php7 | php7.0`` for now. HTH.

**Comment by enoch85** [comment]
@jnweiger I'm not familiar with packiging, but with theese deps, it works:

```
libapache2-mod-php7.0  
php7.0-common  
php7.0-mysql  
php7.0-intl 
php7.0-mcrypt  
php7.0-ldap  
php7.0-imap  
php7.0-cli  
php7.0-gd 
php7.0-pgsql 
php7.0-json  
php7.0-sqlite3 
php7.0-curl  
php7.0-xml  
php7.0-zip 
php-smbclient (could be libsmbclient as well)
```

**Comment by jkroepke** [comment]
Hi,

https://download.owncloud.org/download/repositories/9.0:/testing/xUbuntu_14.04/Packages

```
Package: owncloud-deps-php7
Source: owncloud
Version: 9.0.0~RC3-1.1
Architecture: all
Maintainer: Juergen Weigert <jw@owncloud.com>
Installed-Size: 25
Depends: php7
Provides: owncloud-deps
Filename: ./all/owncloud-deps-php7_9.0.0~RC3-1.1_all.deb
Size: 1536
MD5sum: cd8954914657d404017bc40ce1ec4d76
SHA1: ca2f0f3b2e1e5b9677cfbb34ae10a5255b339b7e
SHA256: 3e635add4a70d21e9134a1845283b44c0364c9cb72bb21be0b9e48303f903199
Section: devel
Priority: optional
Description: ownCloud - dependencies for php7
```

Maybe a cache issue..

**Comment by jkroepke** [comment]
@enoch85

instead depend on libapache2-mod-php7.0, fcgi and fpm, just use depens php7.0 or even php

```
apt-cache showpkg php7.0
Package: php7.0
Versions:
7.0.4-5+deb.sury.org~trusty+1 (/var/lib/apt/lists/ppa.launchpad.net_ondrej_php_ubuntu_dists_trusty_main_binary-amd64_Packages) (/var/lib/dpkg/status)
 Description Language:
                 File: /var/lib/apt/lists/ppa.launchpad.net_ondrej_php_ubuntu_dists_trusty_main_binary-amd64_Packages
                  MD5: 86488d7a405bdcbac9ece7c307f30abb
 Description Language: en
                 File: /var/lib/apt/lists/ppa.launchpad.net_ondrej_php_ubuntu_dists_trusty_main_i18n_Translation-en
                  MD5: 86488d7a405bdcbac9ece7c307f30abb


Reverse Depends:
  owncloud-deps-php7,php7.0
  php,php7.0
Dependencies:
7.0.4-5+deb.sury.org~trusty+1 - php7.0-fpm (16 (null)) libapache2-mod-php7.0 (16 (null)) php7.0-cgi (0 (null)) php7.0-common (0 (null)) php5 (3 5.6.16+dfsg-4~)
Provides:
7.0.4-5+deb.sury.org~trusty+1 - php
Reverse Provides:
```

**Comment by enoch85** [comment]
Ok, did some more tests and I'd say that this fix are legit. Only one change I would prefer though:

1. Do not install `php5-common php5-curl php5-gd php5-intl php5-json php5-mcrypt
  php5-pgsql php5-sqlite` if PHP 7 is recognized.

Thanks for responding on this @jnweiger please release it soon so that we can close this issue.

**Comment by jnweiger** [comment]
@enoch85 thanks for testing the dependencies.
Looks great to me. Only 7.0 packages installed. There is php5-user-cache suggested, but I am sure this is not owncloud related.

The php5 packages are listed as candidates for autoremoval. This is fine. Nothing to do here on my side.

**Comment by jnweiger** [comment]
to be released as part of 9.0.1

**Comment by jkroepke** [comment]
In my opinion, the package has too much dependencies. For example ldap or
samba libs are not required. There should be suggestions instead
dependencies.

Jürgen Weigert <notifications@github.com> schrieb am Fr., 11. März 2016,
17:29:

**Comment by jnweiger** [comment]
I agree. Logic for php7.0 and php5 should be same. changing to:
<pre>
Package: owncloud-deps-php5
Architecture: all
Provides: owncloud-deps
# https://github.com/owncloud/core/issues/21342
Depends: ${misc:Depends}, php5 (>= 5.4.0),
        php5-common, php5-gd, php-xml-parser, php5-intl, php5-mcrypt, php5-curl, php5-json,
        php5-mysqlnd | php5-mysql, php5-sqlite, php5-pgsql,
        php5-cgi | php5-fpm | libapache2-mod-php5,
        apache2 (>= 2.4.0) | httpd
# Don't recommend php5-apcu for Ubuntu-14.04 https://github.com/owncloud/core/issues/14188
# Don't use apcu before 4.0.6 https://github.com/owncloud/core/issues/14175
# php5-imagick is needed by apps/files-versions
# php5-ldap is needed by apps/user-ldap
Suggests: php5-apcu (>= 4.0.6) | php-apc, php5-imagick, php5-ldap
Recommends: mysql-server | mariadb-server | postgresql, curl, exim4 | mail-transport-agent
Description: ownCloud - dependencies for php5
 .

Package: owncloud-deps-php7.0
Architecture: all
Provides: owncloud-deps
# From https://github.com/owncloud/core/issues/20963
Depends: libapache2-mod-php7.0,
        php7.0-common,
        php7.0-mysql,
        php7.0-intl,
        php7.0-mcrypt,
        php7.0-imap,
        php7.0-cli,
        php7.0-gd,
        php7.0-pgsql,
        php7.0-json,
        php7.0-sqlite3,
        php7.0-curl,
        php7.0-xml,
        php7.0-zip
Suggests: php7.0-ldap, php-smbclient | libsmbclient
Description: ownCloud - dependencies for php7

</pre>

Is php7.0-imap needed?

**Comment by enoch85** [comment]
GREAT!!! :D Thanks @jnweiger Do you think you could release this for 9.0 already?

**Comment by jnweiger** [comment]
we are testing and discussing things. That belongs more to testing than to stable.

**Comment by jkroepke** [comment]
If I use nginx as application server, I do not require libapache2. I still
prefer to depends on php7.0 instead libapache2 or fpm..

Jürgen Weigert <notifications@github.com> schrieb am Fr., 11. März 2016,
17:41:

**Comment by jnweiger** [comment]
We can list alternatives there, so that one deps-php7.0 fits bot apache and nginx, 
or we can split into deps-php7.0-apache and deps-php7.0-nginx ...

**Comment by enoch85** [comment]
@jnweiger What I mean is that when the final decision is made, and this issue is closed - then it would be "stable" as it will go in 9.0.1 anyway, right?

**Comment by jnweiger** [comment]
Right. To be released as part of 9.0.1

**Comment by enoch85** [comment]
As this will be released in 9.0.1 I will close this. Feel free to reopen if the issue presists.

**Comment by enoch85** [comment]
This is still an issue: https://github.com/owncloud/documentation/pull/2222#issuecomment-206993826 cc @jnweiger

**Comment by arjanvlek** [comment]
Hi, I used to have PHP 5.6.x on my server, but now I have PHP 7.0 and want to uninstall PHP 5.6 completely.
However, if I try to remove the PHP 5.6 packages, apt-get wants to uninstall owncloud and owncloud-server from my system. How do I uninstall PHP 5.6 while keeping PHP 7.0 and Owncloud 9.0.1?

The only package I could uninstall was libapache2-mod-php5 so there are at least no duplicate PHP modules for apache2 on my server anymore...

**Comment by enoch85** [comment]
@arjanvlek Which OS do you run?

**Comment by arjanvlek** [comment]
@enoch85 Debian 8.4

**Comment by RealRancor** [comment]
@arjanvlek apt-get install owncloud-deps-php7.0 && apt-get remove --purge owncloud-deps-php5

**Comment by enoch85** [comment]
@arjanvlek Just mv the `config` `data` `apps` and `theme` folder outside of your install dir and purge owncloud completely. Then reinstall it and move back the folders.

@RealRancor beat me to it ;)

**Comment by arjanvlek** [comment]
Sorry that it is in Dutch but this is what this command gives me:

De volgende pakketten hebben niet-voldane vereisten:
 owncloud-deps-php7.0 : Vereisten: php7.0-mbstring maar het is niet installeerbaar
                        Vereisten: php7.0-xml maar het is niet installeerbaar
                        Vereisten: php7.0-zip maar het is niet installeerbaar

**Comment by enoch85** [comment]
@arjanvlek What I wrote works anyway, I just did it on 7 servers myself today.

**Comment by arjanvlek** [comment]
It misses some PHP7.0 packages on my server. From which repo did you get php7.0? I used [dotDeb](https://www.dotdeb.org/)

**Comment by arjanvlek** [comment]
@enoch85 Oh wow I see the apt-get command wasn't posted by you! Will try the other method and see if it works.

**Comment by enoch85** [comment]
aptitude purge owncloud* -y
aptitude purge php5* -y
apt-get autoremove -y (I think it's apt-get on Debian)
aptitude install owncloud
and if it doesn't install the php 7 dependencies, then just;
aptitude install owncloud-deps-php7 -y

Done.

**Comment by arjanvlek** [comment]
apt-get install owncloud immediately wants to install php5 packages again...
and that apt-get purge php5* -y command is dangerous as it also removes PHP7!!!

**Comment by enoch85** [comment]
@arjanvlek If you know what you are doing you can just install PHP 7 again. Try to do a `apt-get autoclean` as well, and then restart apache, or the server. Sometimes the servers just don't listen to your commands and you have to force it. ;)

**Comment by arjanvlek** [comment]
Yep I installed PHP7.0 and owncloud again. I found out that PHP7 support is incomplete with dotdeb and therefore owncloud-deps-php7 can't be installed due to missing packages. Therefore I'll leave php5 on my server until I find a better php7 repo. Any suggestions for a better php7 repo? Or wait until debian 9 which will include it in its default repos?

**Comment by enoch85** [comment]
@arjanvlek I use this on Debian: https://github.com/enoch85/ownCloud-VM/blob/master/debian/owncloud_install.sh#L190-214

Good luck.

**Comment by arjanvlek** [comment]
@enoch85 I got everything installed back and enabled php7 using this
command:
`sudo a2dismod php5 && sudo a2enmod php7.0 && sudo service apache2 restart
` The old packages didn't take up that much space and all works fine now.
Also enabled memcache using php7.0-apcu and works great!
Bye
Op 24 apr. 2016 8:01 p.m. schreef "Daniel Hansson" <notifications@github.com

**Comment by PVince81** [comment]
Is this still an issue ?

@owncloud/packaging

**Comment by MrGraversen** [comment]
Personally, I eventually gave up using ownCloud altogether because this issue dragged on for several months. I think it is still an issue.

**Comment by arjanvlek** [comment]
I just left the php5 packages on the system and never looked back. 
Furthermore, I disabled libapache2-mod-php5 and started using php7 only, both the CLI and Apache2 parts of it.
So: While owncloud 9.1.1 stable still "needs" php5 packages installed on Debian 8.5 32 bit, it is actually using php7 (due to libapache2-mod-php7.0 being turned on) and it is running just fine.

**Comment by Vectrex** [comment]
It is no longer a problem. For quite some time it was sufficient to install only the owncloud-files package from the official repository. Running it on Ubuntu 14.04.5 for a few months now without any problems.

**Comment by crrodriguez** [comment]
@Vectrex thanks for letting us to know.

**Comment by working-name** [comment]
Would you mind updating this for https://software.opensuse.org/download.html?project=isv:ownCloud:community&package=owncloud ? It still wants php5. Such an odd thing to do...

**Comment by arjanvlek** [comment]
I confirm this issue is resolved.
All it took me to switch to the correct dependencies was this one-line command:
`sudo apt remove owncloud-deps-php5 && sudo apt install owncloud-deps-php7.0`
Afterwards I removed all remaining php5 packages, such as libapache2-mod-php5 and the base php5 packages. Thanks for fixing it :)
